### ISSUE TITLE
Server Side Validation Messages Not Displaying

### ISSUE THREAD

**Issue by ryanbuening** [saad749/BeginCollectionItemCore#7]
Has anyone tried using this with server side validation? I put a [Required] attribute on one of my properties in my model class and when I go to submit (with client side validation disabled) I can see that the ModelState is not valid. However, the required message does not get added to the view.

If I keep client side validation enabled, the required message does show.

**Comment by ryanbuening** [comment]
I've uploaded my sample to GitHub: https://github.com/ryanbuening/DynamicCollections

**Comment by ryanbuening** [comment]
Similar issue (and possible solution) is described at this link towards the middle of the page: http://ivanz.com/2011/06/16/editing-variable-length-reorderable-collections-in-asp-net-mvc-part-1/

**Comment by ryanbuening** [comment]
### I've copy/pasted the relevant information from the article I referenced above. I'm new to .NET and C# so I'm not sure if this is a viable solution, but it sounded very similar.
----------------------------------------
...It’s the same problem I mentioned earlier in this post. Each time we render the view we assign different names to the fields, which do not match the ones submitted and leads to a ModelState inconsistency. We have to figure out how to persist the name and more specifically the Index across requests. We have two options:

1. Add a hidden CollectionIndex field and CollectionIndex property on the object to persist the Index. This however is intrusive and suboptimal.
2. Instead of polluting the object with an extra property be smart and in our helper Html.BeginCollectionItem reapply/reuse the submitted Index form values.

Let’s replace in Html.BeginCollectionItem this line:
`string itemIndex = Guid.New().ToString();`
with:
`string itemIndex = GetCollectionItemIndex(collectionIndexFieldName);`

And here’ is the code for GetCollectionItemIndex:
```
private static string GetCollectionItemIndex(string collectionIndexFieldName)
{
    Queue<string> previousIndices = (Queue<string>) HttpContext.Current.Items[collectionIndexFieldName];
    if (previousIndices == null) {
        HttpContext.Current.Items[collectionIndexFieldName] = previousIndices = new Queue<string>();

        string previousIndicesValues = HttpContext.Current.Request[collectionIndexFieldName];
        if (!String.IsNullOrWhiteSpace(previousIndicesValues)) {
            foreach (string index in previousIndicesValues.Split(','))
                previousIndices.Enqueue(index);
        }
    }

    return previousIndices.Count > 0 ? previousIndices.Dequeue() : Guid.NewGuid().ToString();
}
```

We get all submitted values, put them in a queue, which we store for the duration of the request. Each time we render a collection item we dequeue its old index value and if none is available we generate a new one. That way we preserve the Index across requests and can have a consistent ModelState and see validation errors and messages.

**Comment by ryanbuening** [comment]
@saad749 - Do you think this bug fixable? Server side validation is a requirement for the app I'm building and I'm trying to decide whether or not I should continue using BeginCollectionItem or if I need to look for alternative methods. Not in any rush...just checking to see if you think this bug can be fixed in theory. Thanks.

**Comment by saad749** [comment]
Sorry, I have been very busy with other projects, and didn't had the time to look into it in detail. If you have been able to find a fix for this, you can send it as a PR.

**Comment by ryanbuening** [comment]
@saad749 No problem. I completely understand. It looks like @marcioluizsf created a PR that might solve this bug: https://github.com/saad749/BeginCollectionItemCore/pull/8

**Comment by saad749** [comment]
Yes. Thanks to PR from @marcioluizsf . I have published version 1.0.8 with the fix.

**Comment by marcioluizsf** [comment]
I'm very happy to have contributed.
Thank you @saad749 for the new version.
